scalanet/src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -302,11 +310,11 @@ class UDPPeerGroup[M](val config: Config)(
       }
   }
 
-  override def server(): ConnectableObservable[ServerEvent[InetMultiAddress, M]] = serverSubject
+  override def server(): Observable[ServerEvent[InetMultiAddress, M]] =
+    Observable.repeatEvalF(serverQueue.poll).cache
```

Comment:
Hm, doesn't `.cache` mean that from the time of the first subscription it caches all elements so that it can replay it to the next subscriber? What's the need for this here? Why cache all server events such as new connections coming in? Won't that result in a memory leak?

Reply:
yep you are right, i have probably experimented with it and forgot to delete.